src/components/RegisterForm.js

```diff
@@ -1,59 +1,273 @@
+  };
+  const pincodeChangeHandler = (e) => {
+    setSuccMsg("");
+    setPinCodeError("");
+    setPinCode(e.target.value);
+  };
+  const genderChangeHandler = (e) => {
+    setSuccMsg("");
+    setGenderError("");
+    setGender(e.target.value);
+  };
+  const addressChangeHandler = (e) => {
+    setSuccMsg("");
+    setAddressError("");
+    setAddress(e.target.value);
+  };
+  const submitTimeHandler = (e) => {
+    e.preventDefault();
+    if (firstname !== "") {
+      if (lastname !== "") {
+        if (email !== "") {
+          const emailRegex = /^\w+([-+.']\w+)*@\w+([-.]\w+)*\.\w+([-.]\w+)*$/;
+          if (emailRegex.test(email)) {
+            if (mobile !== "") {
+              if (mobile.trim().length === 10) {
+                if (username !== "") {
+                  const userRegex = /^[a-zA-Z0-9]+$/;
+                  if (userRegex.test(username)) {
+                    if (dob !== "") {
+                      if (pincode !== "") {
+                        if (address !== "") {
+                          setSuccMsg("you are successfully logged in!!");
+                        } else {
+                          setAddressError("address required");
+                        }
+                      } else {
+                        setPinCodeError("pincode required");
+                      }
+                    } else {
+                      setDobError(" DOB required");
+                    }
+                  } else {
+                    setUserNameError("not a valid useranme");
+                  }
+                } else {
+                  setUserNameError(" username required");
+                }
+              } else {
+                setMobileError("Invalid Mobile Number");
+              }
+            } else {
+              setMobileError("Mobile number required");
+            }
+          } else {
+            setEmailError("not a valid email id");
+          }
+        } else {
+          setLastNameError("email required");
+        }
+      } else {
+        setLastNameError("lastname required");
+      }
+    } else {
+      setFirstNameError("firstname required");
+    }
+    const registerData = {
+      fname: firstname,
+      lname: lastname,
+      emailId: email,
+      mobilenum: mobile,
+      userName: username,
+      DateOfBirth: dob,
+      pin: pincode,
+      Gender: gender,
+      Address: address,
+    };
+    console.log(registerData);
+  };
+
+  return (
+    <form onSubmit={submitTimeHandler}>
+      <div className="form-inner">
+        <h2>Register Here!</h2>
+        {succMsg && <div className="success-msg">{succMsg}</div>}
+        <div className="form-group">
+          <label htmlFor="firstname">First Name:</label>
+          <input
+            type="text"
+            name="firstname"
+            id="firstname"
+            placeholder="enter your first Name"
+            value={firstname}
+            onChange={fisrtnameChangeHandler}
+          />
```

Comment:
You can recreate one reusable component for TextField, that accepts label, type, name, id, placeholder etc. as props.

So by adding TextField you can add any no. of same input fields. With this we will have control in one single component.